src/lib/y2partitioner/ui_state.rb

```diff
@@ -19,22 +19,16 @@
 
 module Y2Partitioner
   # Singleton class to keep the position of the user in the UI and other similar
-  # information that needs to be rememberd across UI redraws to give the user a
+  # information that needs to be remembered across UI redraws to give the user a
   # sense of continuity.
   class UIState
-    include Yast::I18n
-
-    # Constructor
+    # A collection holding a PageStatus for each CWM::Page visited by the user
     #
-    # Called through {.create_instance}, starts with a blank situation (which
-    # means default for each widget will be honored).
-    def initialize
-      textdomain "storage"
-
-      @candidate_nodes = []
-      @open_items = {}
-      @overview_tree_pager = nil
-    end
+    # The CWM::Page#widget_id is used as index
+    # @see #status_for
+    #
+    # @return [Hash{String => PageStatus}]
```

Comment:
Why a hash? I would say we can use an array of PageStatus. We can retrieve a PageStatus by the page id.

Reply:
Yes, you are right.

But, after having a look, I'd keep the current approach because it looks more performant to me. Changing to a not indexed structure force us to iterates over it until find the desired PageStatus.

Do you agree?

Reply:
Umm, I would prefer a plain array. It is easier to understand, and there will not be performance issues in that find actions. Only in case of having hundreds of pages we would notice some difference, otherwise such performance penalty would not be noticeable.

Reply:
I'm partially agree with you. Let's go for a plain array.

Thanks!